sirepo/pkcli/job_supervisor.py

```diff
@@ -106,12 +114,49 @@ def sr_on_exception(self):
         self.close()
 
 
+# Must be defined before use
+def _db_file_req_validated(func):
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        t = self.request.headers.get('Authorization')
+        if not t:
+            raise tornado.web.HTTPError(401)
+        p = t.split(' ')
+        if len(p) != 2:
+            raise tornado.web.HTTPError(401)
+        if p[0] != 'Bearer' or not sirepo.job.UNIQUE_KEY_RE.search(p[1]):
+            raise tornado.web.HTTPError(401)
+        if p[1] not in _DB_FILE_KEYS or \
+           self.request.path.split('/')[3] != _DB_FILE_KEYS[p[1]]:
+            raise tornado.web.HTTPError(403)
+        # TODO(e-carlin): discuss with rn what validation needs to happen
+        p = self.request.path
+        if p.count('.') > 1:
+            raise tornado.web.HTTPError(404)
+        return func(self, *args, **kwargs)
+    return wrapper
+
+
+# TODO(e-carlin): Is serving large files like this going to be slow?
```

Comment:
@robnagler from the little bit I've read Tornado is "slow" (ex compared to NGINX) for serving large files and can sometimes consumes a great deal of RAM. Should we worry about this?

Reply:
cde21fe5